tools/pyboard.py

```diff
@@ -123,7 +123,25 @@ def __init__(self, device, baudrate=115200, user='micro', password='python'):
             self.serial = TelnetToSerial(device, user, password, read_timeout=10)
         else:
             import serial
-            self.serial = serial.Serial(device, baudrate=baudrate, interCharTimeout=1)
+            delayed = False
+            delay = 25
+            for attempt in range(delay):
+                try:
+                    self.serial = serial.Serial(device, baudrate=baudrate, interCharTimeout=1)
+                    done = True
+                    break
+                except serial.serialutil.SerialException:
```

Comment:
"serial.serialutil.SerialException" looks like a private class, isn't it re-exported under simpler name. And maybe it's OSError subclass after all?